utils/data-loading.ts

```diff
@@ -15,12 +16,26 @@ export function useDataSource<Data, Deps extends DependencyList | undefined, E e
   const [isLoading, setIsLoading] = useState<boolean>(false)
   const [data, setData] = useState<Data | null>(null)
   const [error, setError] = useState<E | null>(null)
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const reloadRef = useRef<() => void>(_.noop)
+
   useEffect(() => {
-    setIsLoading(true)
-    createDataPromise()
-      .then(setData)
-      .catch(setError)
-      .finally(() => setIsLoading(false))
+    // The isSubscribed logic is necessary to prevent setStates after unmounts or dependency changes
+    let isSubscribed = true
+    // For isSubscribed to work with reloading we need to use reload as a Ref
+    reloadRef.current = () => {
+      isSubscribed && setIsLoading(true)
```

Comment:
I think this would read better a spelled-out `if`:

```suggestion
      if (isSubscribed) {
        setIsLoading(true)
      }
````

Reply:
I have resolved this by returning if not subscribed which simplifies it out.

Reply:
Actually after writing a test for it I am realising checking for `isSubscribed` here is completely unnecessary, since caller has a ref they will always have the latest version of this function and `isSubscribed = false => reloadRef.current = noop`.